Scaffold config new template

rin-st opened the pull request:
to test:

```
yarn cli -e rin-st/scaffold-config-test
```

note: resulting config doesn't have comments like in se-2 or this create-eth template

Result:

```
import * as chains from "viem/chains";

export type ScaffoldConfig = {
  targetNetworks: readonly chains.Chain[];
  pollingInterval: number;
  alchemyApiKey: string;
  walletConnectProjectId: string;
  onlyLocalBurnerWallet: boolean;
};

export const DEFAULT_ALCHEMY_API_KEY = "oKxs-03sij-U_N0iOlrSsZFr29-IqbuF";

// Custom variables
const CUSTOM_API_KEY = process.env.CUSTOM_API_KEY;

const scaffoldConfig: ScaffoldConfig = {
  alchemyApiKey: process.env.NEXT_PUBLIC_ALCHEMY_API_KEY || DEFAULT_ALCHEMY_API_KEY,
  walletConnectProjectId: process.env.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID || "3a8170812b534d0ff9d794f19a901d64",
  targetNetworks: [chains.hardhat, chains.base],
  pollingInterval: 12345,
  onlyLocalBurnerWallet: false,
};

export default scaffoldConfig;
```

extension: https://github.com/rin-st/scaffold-config-test

carletex commented on templates/solidity-frameworks/foundry/packages/nextjs/scaffold.config.ts.args.mjs:
```diff
@@ -1 +1,3 @@
-export const chainName = 'foundry'
\ No newline at end of file
+export const configOverrides = {
+  targetNetworks: ["$$chains.foundry$$"]
```
Oh damn, this is how people need to use it, right?

Can you explain in which cases this is needed? Sorry, I'm sure we've talked about this a bunch of times.

rin-st replied:
> Oh damn, this is how people need to use it, right?

Yep

> Can you explain in which cases this is needed?

We need to use it when we want the result (variable or expression) to be used by js runtime. In our case we want the result: `targetNetworks: [chains.foundry]` where `chains.foundry` is a variable used in runtime.

We can't use `targetNetworks: ["chains.foundry"]` or `targetNetworks: "[chains.foundry]"` in template because result will be the same (`targetNetworks: ["chains.foundry"]` or `targetNetworks: "[chains.foundry]"` )

We added info about it [here](https://github.com/scaffold-eth/create-eth/blob/beta/contributors/TEMPLATING.md#rules-for-template-args)

carletex replied:
Ok, thanks! I didn't look at the PR closely and had the impression that we only needed to use those "wildcards" on the template file, not the template `args`.

It's ugly as hell, but I guess we explored a bunch of options and this was the best.

technophile-04 reviewed (commented):
Thanks Rinat 🙏!! But I think the way we added new args doesn't provide full customizability also loosing those comments feels bad :( 

So like we need this two main things: 

1. Adding new properties to scaffold config object.
    - currently we can't do it because there is no way to extend types and even though we pass extra properties in `configOverride` it error out because of types constrain `ScaffoldConfig`
2. Not loosing the current comments.  

So to tackle 1 & 2: 

1. We have `type baseConfig` and merge that with extension type config to form final `ScaffoldConfig` type. 
2. For comments, instead of having on them object let's move them inside the types? 

Something like this: 

<details>

<summary>
    scaffold.config.ts.template.mjs
</summary>

```js
import { withDefaults, stringify, deepMerge } from "../../../../templates/utils.js";

const defaultScaffoldConfig = {
  targetNetworks: ["$$chains.mainnet$$"],
  pollingInterval: 30000,
  alchemyApiKey: "$$process.env.NEXT_PUBLIC_ALCHEMY_API_KEY || DEFAULT_ALCHEMY_API_KEY$$",
  walletConnectProjectId: "$$process.env.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID || '3a8170812b534d0ff9d794f19a901d64'$$",
  onlyLocalBurnerWallet: true,
};

const contents = ({ preConfigContent, configOverrides, configTypeName }) => {
  // add solidityFramework network
  const targetNetworks = configOverrides.map(override => override.targetNetworks).flat();
  const extensionConfigOverrides = configOverrides[configOverrides.length - 1] || {};
  if (targetNetworks?.length && Object.keys(extensionConfigOverrides).length > 0) {
    extensionConfigOverrides.targetNetworks = targetNetworks;
  }

  // Merge the default config with any overrides
  const finalConfig = deepMerge(defaultScaffoldConfig, extensionConfigOverrides);

  return  `import * as chains from "viem/chains";
${preConfigContent[0] || ''}

export type baseConfig = {
  // The networks on which your DApp is live
  targetNetworks: readonly chains.Chain[];
  
  // The interval at which your front-end polls the RPC servers for new data
  // it has no effect if you only target the local network (default is 4000)
  pollingInterval: number;
  
  // This is ours Alchemy's default API key.
  // You can get your own at https://dashboard.alchemyapi.io
  // It's recommended to store it in an env variable:
  // .env.local for local testing, and in the Vercel/system env config for live apps.
  alchemyApiKey: string;
  
  // This is ours WalletConnect's default project ID.
  // You can get your own at https://cloud.walletconnect.com
  // It's recommended to store it in an env variable:
  // .env.local for local testing, and in the Vercel/system env config for live apps.
  walletConnectProjectId: string;
  
  // Only show the Burner Wallet when running on hardhat network
  onlyLocalBurnerWallet: boolean;
};

export type ScaffoldConfig = baseConfig ${configTypeName[0] ? `& ${configTypeName[0]}` : ''};

export const DEFAULT_ALCHEMY_API_KEY = "oKxs-03sij-U_N0iOlrSsZFr29-IqbuF";

const scaffoldConfig: ScaffoldConfig = ${stringify(finalConfig)};

export default scaffoldConfig;`;
};

export default withDefaults(contents, {
  preConfigContent: "",
  configOverrides: {},
  configTypeName: "",
});
```

</details>

<details>

<summary>
    scaffold.config.ts.args.mjs
</summary>

```js
export const preConfigContent = `
// Custom variables
const CUSTOM_API_KEY = process.env.CUSTOM_API_KEY;

export type ExtraConfig = {
  // Random comment
  customApiKey: string | undefined;
}`;

export const configTypeName = "ExtraConfig";

export const configOverrides = {
  targetNetworks: ["$$chains.base$$"],
  pollingInterval: 12_345,
  onlyLocalBurnerWallet: false,
  customApiKey: "$$CUSTOM_API_KEY$$",
};
```

</details>

But maybe I am missing and there might be simpler approach and would love to know it! 

Also lol we have discovered a case `types` as args, maybe in above example I passs them as string but not sure if ihave any other better approach?

rin-st commented:
Great points @technophile-04 ! Updated PR

Regarding types I thought about this options:
- hardcode type name, something like `AdditionalExtensionScaffoldConfigs` so we don't need to pass it. But user could occasionally change it in predefined config and everything can break
- use `typeof keyof configOverrides` as additional type. But in this case that new type can have intersections with `BaseConfig` if something from default config is overridden

So for now I left it as you proposed. We can think about this later if we need to work more with passing types

Thanks!

technophile-04 commented:
Thanks @rin-st! Looks good! Also, can we make a PR to `create-eth-extensions`  branch? I have just created the [`example-beta`](https://github.com/scaffold-eth/create-eth-extensions/tree/example-beta) maybe we keep updating that branch when we update the `.template.mjs` and finally when we are ready we merge `example-beta` => `example`


Also just cc @carletex on his views on https://github.com/scaffold-eth/create-eth/pull/227#pullrequestreview-2666766523

technophile-04 commented on templates/base/packages/nextjs/scaffold.config.ts.template.mjs:
```diff
@@ -1,45 +1,62 @@
+export type ScaffoldConfig = BaseConfig ${extraConfigTypeName[0] ? `& ${extraConfigTypeName[0]}` : ''};
+
+export const DEFAULT_ALCHEMY_API_KEY = "oKxs-03sij-U_N0iOlrSsZFr29-IqbuF";
+${preConfigContent[0] || ''}
+;
```
nit: I think we shall move `preConfigContent` above `BaseConfig` so that if people want they can also add imports? Like importing custom chain from other file.  I mean it should work because of prettier will format it later but maybe its make sense to have it above all. 

Also there is extra semicolon on line 51

technophile-04 replied:
updated at https://github.com/scaffold-eth/create-eth/pull/227/commits/b899ec781965351f12ab9cc281bd7c6f653934c5 

And also move the type to be `as const Satisfies` so that we don't loose literal numbers

technophile-04 commented:
> Could you explain what you expect to get and what are you getting as a result. For me it seems its working as expected

Ohh ohh my bad, lol I had lot of branches while testing and didn't push the multichain args commit to https://github.com/scaffold-eth/create-eth-extensions/tree/beta-scaffold-config but now it working great 🙌

carletex commented:
Regarding losing comments on deepMerge. I'm here with Shiv and Cursor and we came with a (hacky) solution if we wanted to use it.

Change utils `stringify` function to (compact = false + magic comment placement):
```js
export const stringify = (val, comments = {}) => {
  let str = inspect(val, { 
    depth: null, 
    compact: false,
    maxArrayLength: null, 
    maxStringLength: null
  });
  
  // Add comments above their respective properties
  if (Object.keys(comments).length > 0) {
    str = str.replace(/^\s*(\w+):/gm, (match, prop) => {
      const indent = match.match(/^\s*/)[0];
      return comments[prop] 
        ? `${indent}// ${comments[prop]}\n${match}`
        : match;
    });
  }

  return str
    .replace(/"\$\$([^"]+)\$\$"/g, '$1')
    .replace(/'\$\$([^']+)\$\$'/g, '$1')
    .replace(/(['"])(.*?\$\{.*?\}.*?)\1/g, '`$2`');
};

```

And you could use it in the template like:

```js
const scaffoldConfig = ${stringify(finalConfig, {
  targetNetworks: "The networks on which your DApp is live",
  pollingInterval: "The interval at which your front-end polls the RPC servers for new data",
  alchemyApiKey: "Your Alchemy API key for interacting with Ethereum networks",
  walletConnectProjectId: "Your WalletConnect project ID for connecting to mobile wallets",
  onlyLocalBurnerWallet: "Only show the Burner Wallet when running on hardhat network"
})} as const satisfies ScaffoldConfig;
```

(fake ai comments lol)

We could also made it multi-line adding (\n).

A bit hacky, but it would solve the issue if we wanted.

carletex commented:
Another thing we need to fix is the sorting of the props

This is what we are getting in the instance (using the logic from the previous comment, but it's an existing problem)
![image](https://github.com/user-attachments/assets/be9b3012-e453-40e4-98ba-f3382715e532)

Deepmerge need to respect the order, or we could do another hacky solution :D

rin-st commented:
> Deepmerge need to respect the order, or we could do another hacky solution :D

Updated `deepMerge` so it now respects the first argument's keys order.

> Regarding losing comments on deepMerge. I'm here with Shiv and Cursor and we came with a (hacky) solution if we wanted to use it.

For me, comments in types are enough and I think less hacky solutions -> better. But if you think we need it I'll add it

carletex commented:
> For me, comments in types are enough

The problem might be in files like hardhat config, right? Where we don't have the types.

Anyway, we can always add it (the possibility to add comments on objects) at some point if we feel we need it.

PS. I don't think the solution was super hacky (just a bit :)).

rin-st commented:
> The problem might be in files like hardhat config, right? Where we don't have the types.

Agree, good point!

> Anyway, we can always add it (the possibility to add comments on objects) at some point if we feel we need it.

I updated `stringify` a bit so it works with multiline comments and and added it to scaffold config

But `stringify` needs to be updated later. It finds all the keys with the passed name in the object in any depth, and add comments to it. So if you have an object 
```
{
  // comment for network [1]
  network: {
      ...
  },
  something: 
    {
       // same comment [1] will be there
       network: ...
    }
}
```
So we need the way to find exact place where add to comment. For example for second to use `network` for the first and `something.network` for second

technophile-04 commented:
Thanks @rin-st for this!! Work nicely, can we update it so that we pick `parent.child` relation? or the logic is getting messier with it?

rin-st commented:
> Work nicely, can we update it so that we pick parent.child relation? or the logic is getting messier with it?

A bit messier but we can live with it since we need that functionality. Before merge I'll add changes to example extension + maybe info about adding comments

---

- formatted utils file
- added deep level comments support
- added new field`rpcOverrides` to default scaffold.config and also handling comments inside empty object for it (Example part):
```
  // If you want to use a different RPC for a specific network, you can add it here.
  // The key is the chain ID, and the value is the HTTP RPC URL
  rpcOverrides: {
    // Example:
    // [chains.mainnet.id]: "https://mainnet.buidlguidl.com",
  },

```
To make it work I added possibility to add comments ending with a dot: `"rpcOverrides.": "Example..."`

---

To test:

Example `scaffold.config.ts.args.mjs` file:

<details><summary>Details</summary>
<p>

```
export const extraConfigTypeName = "ExtraConfig";

export const preConfigContent = `
// Custom variables
const CUSTOM_API_KEY = process.env.CUSTOM_API_KEY;

export type ${extraConfigTypeName} = {
  // Random comment
  customApiKey: string | undefined;
  something: any;
}
`;

export const configOverrides = {
  targetNetworks: ["$$chains.base$$", "$$chains.baseSepolia$$"],
  pollingInterval: 12_345,
  onlyLocalBurnerWallet: false,
  customApiKey: "$$CUSTOM_API_KEY$$",
  something: {
    targetNetworks: ["$$chains.baseSepolia$$", "$$chains.base$$"],
    somethingElse: {
      a: "no-comments-here",
      b: "b",
      c: 4,
      targetNetworks: ["$$chains.base$$"],
    }
  }
};

```

</p>
</details> 

`stringify` with comments from `scaffold.config.ts.template.mjs`

<details><summary>Details</summary>
<p>

```
const scaffoldConfig = ${stringify(finalConfig, {
  targetNetworks: "The networks on which your DApp is live",
  pollingInterval: "The interval at which your front-end polls the RPC servers for new data (it has no effect if you only target the local network (default is 4000))",
  alchemyApiKey: "This is ours Alchemy's default API key.\nYou can get your own at https://dashboard.alchemyapi.io\nIt's recommended to store it in an env variable:\n.env.local for local testing, and in the Vercel/system env config for live apps.",
  walletConnectProjectId: "This is ours WalletConnect's default project ID.\nYou can get your own at https://cloud.walletconnect.com\nIt's recommended to store it in an env variable:\n.env.local for local testing, and in the Vercel/system env config for live apps.",
  rpcOverrides: "If you want to use a different RPC for a specific network, you can add it here.\nThe key is the chain ID, and the value is the HTTP RPC URL",
  "rpcOverrides.": "Example:\n[chains.mainnet.id]: \"https://mainnet.buidlguidl.com\",",
  onlyLocalBurnerWallet: "Only show the Burner Wallet when running on hardhat network",
  "something.targetNetworks": "something.targetNetworks comment",
  "something.somethingElse.b": "something.somethingElse.b comment",
  "something.somethingElse.c": "something.somethingElse.c comment",
})} as const satisfies ScaffoldConfig;
```

</p>
</details> 


Result: 

<details><summary>Details</summary>
<p>

```
import * as chains from "viem/chains";


// Custom variables
const CUSTOM_API_KEY = process.env.CUSTOM_API_KEY;

export type ExtraConfig = {
  // Random comment
  customApiKey: string | undefined;
  something: any;
}


export type BaseConfig = {
  targetNetworks: readonly chains.Chain[];
  pollingInterval: number;
  alchemyApiKey: string;
  rpcOverrides?: Record<number, string>;
  walletConnectProjectId: string;
  onlyLocalBurnerWallet: boolean;
};

export type ScaffoldConfig = BaseConfig & ExtraConfig;

export const DEFAULT_ALCHEMY_API_KEY = "oKxs-03sij-U_N0iOlrSsZFr29-IqbuF";

const scaffoldConfig = {
  // The networks on which your DApp is live
  targetNetworks: [
    chains.hardhat,
    chains.base,
    chains.baseSepolia
  ],
  // The interval at which your front-end polls the RPC servers for new data (it has no effect if you only target the local network (default is 4000))
  pollingInterval: 12345,
  // This is ours Alchemy's default API key.
  // You can get your own at https://dashboard.alchemyapi.io
  // It's recommended to store it in an env variable:
  // .env.local for local testing, and in the Vercel/system env config for live apps.
  alchemyApiKey: process.env.NEXT_PUBLIC_ALCHEMY_API_KEY || DEFAULT_ALCHEMY_API_KEY,
  // If you want to use a different RPC for a specific network, you can add it here.
  // The key is the chain ID, and the value is the HTTP RPC URL
  rpcOverrides: {
    // Example:
    // [chains.mainnet.id]: "https://mainnet.buidlguidl.com",
  },
  // This is ours WalletConnect's default project ID.
  // You can get your own at https://cloud.walletconnect.com
  // It's recommended to store it in an env variable:
  // .env.local for local testing, and in the Vercel/system env config for live apps.
  walletConnectProjectId: process.env.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID || '3a8170812b534d0ff9d794f19a901d64',
  // Only show the Burner Wallet when running on hardhat network
  onlyLocalBurnerWallet: false,
  customApiKey: CUSTOM_API_KEY,
  something: {
    // something.targetNetworks comment
    targetNetworks: [
      chains.baseSepolia,
      chains.base
    ],
    somethingElse: {
      a: 'no-comments-here',
      // something.somethingElse.b comment
      b: 'b',
      // something.somethingElse.c comment
      c: 4,
      targetNetworks: [
        chains.base
      ]
    }
  }
} as const satisfies ScaffoldConfig;

export default scaffoldConfig;
```

</p>
</details>

technophile-04 commented:
Ohh cool been testing for a while and it also added the nicely here here too : 

<details>

<summary>
    Exmaple `scaffold.config.ts.template.mjs` file
</summary>

```js
import { withDefaults, stringify, deepMerge } from "../../../../templates/utils.js";

const defaultScaffoldConfig = {
    targetNetworks: ["$$chains.mainnet$$"],
    pollingInterval: 30000,
    alchemyApiKey: "$$process.env.NEXT_PUBLIC_ALCHEMY_API_KEY || DEFAULT_ALCHEMY_API_KEY$$",
    rpcOverrides: {},
    walletConnectProjectId: "$$process.env.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID || '3a8170812b534d0ff9d794f19a901d64'$$",
    onlyLocalBurnerWallet: true,
    advanced: {
      networking: {
        protocols: {
          http: {
            settings: {
              timeout: 5000,
              retries: 3,
              backoff: {
                initial: 1000,
                factor: 2,
                maxDelay: 10000
              }
            }
          }
        },
        security: {
          encryption: {
            enabled: true,
            algorithm: "AES-256-GCM"
          }
        }
      },
      monitoring: {
        metrics: {
          enabled: true,
          interval: 60000
        }
      }
    }
  };

const contents = ({ preConfigContent, configOverrides, extraConfigTypeName }) => {
  // add solidityFramework network
  const targetNetworks = configOverrides.map(override => override.targetNetworks).flat();
  const extensionConfigOverrides = configOverrides[configOverrides.length - 1] || {};
  if (targetNetworks?.length && Object.keys(extensionConfigOverrides).length > 0) {
    extensionConfigOverrides.targetNetworks = targetNetworks;
  }

  // Merge the default config with any overrides
  const finalConfig = deepMerge(defaultScaffoldConfig, extensionConfigOverrides);

  return  `import * as chains from "viem/chains";

${preConfigContent[0] || ''}

export type BaseConfig = {
  targetNetworks: readonly chains.Chain[];
  pollingInterval: number;
  alchemyApiKey: string;
  rpcOverrides?: Record<number, string>;
  walletConnectProjectId: string;
  onlyLocalBurnerWallet: boolean;
  advanced: {
    networking: {
      protocols: {
        http: {
          settings: {
            timeout: number;
            retries: number;
            backoff: {
              initial: number;
              factor: number;
              maxDelay: number;
            };
          };
        };
      };
      security: {
        encryption: {
          enabled: boolean;
          algorithm: string;
        };
      };
    };
    monitoring: {
      metrics: {
        enabled: boolean;
        interval: number;
      };
    };
  };
};

export type ScaffoldConfig = BaseConfig ${extraConfigTypeName[0] ? `& ${extraConfigTypeName[0]}` : ''};

export const DEFAULT_ALCHEMY_API_KEY = "oKxs-03sij-U_N0iOlrSsZFr29-IqbuF";

const scaffoldConfig = ${stringify(finalConfig, {
  targetNetworks: "The networks on which your DApp is live",
  pollingInterval: "The interval at which your front-end polls the RPC servers for new data (it has no effect if you only target the local network (default is 4000))",
  alchemyApiKey: "This is ours Alchemy's default API key.\nYou can get your own at https://dashboard.alchemyapi.io\nIt's recommended to store it in an env variable:\n.env.local for local testing, and in the Vercel/system env config for live apps.",
  walletConnectProjectId: "This is ours WalletConnect's default project ID.\nYou can get your own at https://cloud.walletconnect.com\nIt's recommended to store it in an env variable:\n.env.local for local testing, and in the Vercel/system env config for live apps.",
  rpcOverrides: "If you want to use a different RPC for a specific network, you can add it here.\nThe key is the chain ID, and the value is the HTTP RPC URL",
  "rpcOverrides.": "Example:\n[chains.mainnet.id]: \"https://mainnet.buidlguidl.com\",",
  "advanced.": "Advanced configuration options for fine-tuning the DApp",
  "advanced.networking.": "Network-related configuration settings",
  "advanced.networking.protocols.http.": "HTTP protocol specific settings",
  "advanced.networking.protocols.http.settings.": "HTTP client configuration",
  "advanced.networking.protocols.http.settings.timeout": "Timeout in milliseconds for HTTP requests",
  "advanced.networking.protocols.http.settings.retries": "Number of retry attempts for failed requests",
  "advanced.networking.protocols.http.settings.backoff.": "Exponential backoff configuration for retries",
  "advanced.networking.protocols.http.settings.backoff.initial": "Initial delay in milliseconds before first retry",
  "advanced.networking.protocols.http.settings.backoff.factor": "Multiplication factor for subsequent retry delays",
  "advanced.networking.protocols.http.settings.backoff.maxDelay": "Maximum delay in milliseconds between retries",
  "advanced.networking.security.": "Security-related settings",
  "advanced.networking.security.encryption.": "Data encryption configuration",
  "advanced.networking.security.encryption.enabled": "Whether to enable encryption for sensitive data",
  "advanced.networking.security.encryption.algorithm": "The encryption algorithm to use",
  "advanced.monitoring.": "System monitoring configuration",
  "advanced.monitoring.metrics.": "Metrics collection settings",
  "advanced.monitoring.metrics.enabled": "Whether to enable metrics collection",
  "advanced.monitoring.metrics.interval": "Interval in milliseconds between metrics collection"
})} as const satisfies ScaffoldConfig;

export default scaffoldConfig;`;
};

export default withDefaults(contents, {
  preConfigContent: "",
  extraConfigTypeName: "",
  configOverrides: {},
});
```

</details>

technophile-04 commented on templates/utils.js:
```diff
@@ -2,45 +2,195 @@ import { inspect } from "util";
+  });
+
+  // Add comments above their respective properties
+  if (Object.keys(comments).length > 0) {
```
Maybe we do early return? 

```
if (Object.key(comments).length === 0) return str;
``` 

So the the code easy parse?

rin-st replied:
thanks, updated [b9bc75a](https://github.com/scaffold-eth/create-eth/pull/227/commits/b9bc75a14f2858e6fe6d196221d3cf4e81c9392b)

rin-st commented:
> Ohh cool been testing for a while and it also added the nicely here here too :

Great! I'll add it to the example extension after approve (or you can add it yourself)

technophile-04 commented on templates/utils.js:
```diff
@@ -2,45 +2,198 @@ import { inspect } from "util";
+  Object.keys(comments)
+    .filter(key => !key.includes("."))
+    .forEach(prop => {
+      const regex = new RegExp(`(^|\\n)(\\s*)(${prop}):\\s`, "g");
```
Nit: Maybe we also name the regexes nicely? 

```js
  // For top-level properties
  const findTopLevelPropertyPattern = new RegExp(`(^|\\n)(\\s*)(${prop}):\\s`, "g");
  // Matches: "\n  propertyName: "

  // For finding object start
  const findObjectStartPattern = new RegExp(`(^|\\n)(\\s*)(${currentPath}):\\s*\\{`, "g");
  // Matches: "\n  objectName: {"

  // For nested properties
  const findNestedPropertyPattern = new RegExp(`(\\n)(${expectedIndent})(${part}):\\s`, "g");
  // Matches: "\n    nestedProperty: "

```

rin-st replied:
Yes, thanks! Updated, but named a bit different. `find` word looks redundant since regexes used mostly for finding, and changed `Pattern` to `Regex` to be more explicit [cc9a3a7](https://github.com/scaffold-eth/create-eth/pull/227/commits/cc9a3a734c10d004501b522fa24ee83d7e9db989)

technophile-04 replied:
Make more sense 🙌

technophile-04 reviewed (approved):
Tysm @rin-st!! Looking nice! Just added last nitpick but looks good to me after chatting with AI and understanding the logic nicely feel confident 🙌

> Great! I'll add it to the example extension after approve (or you can add it yourself)

Ohh no no I think the extension looks good 🙌, it just was for testing
